### ISSUE TITLE
Encountering psycopg.errors.InvalidSqlStatementName, but only sometimes

### ISSUE THREAD

**Issue by jfilby** [psycopg/psycopg#280]
Sometimes my SQL queries raise an exception, here's an example of the tail-end of such an exception:

```
  File "/home/ubuntu/.local/lib/python3.8/site-packages/psycopg/cursor.py", line 555, in execute
    raise ex.with_traceback(None)
psycopg.errors.InvalidSqlStatementName: prepared statement "_pg3_10" does not exist
```

Would upgrading my Python version help? The same queries often work.

**Comment by dvarrazzo** [comment]
Hello,

no, it seems an error in psycopg 3 related to prepared statements.

What server are you using? A normal postgres, connected directly, or some other implementation (AWS) or are you using middleware such as pgbouncer?

You can disable the use of prepared statements by setting [`Connection.prepared_threshold  = None`](https://www.psycopg.org/psycopg3/docs/api/connections.html#psycopg.Connection.prepare_threshold).

**Comment by jfilby** [comment]
Yes, I'm using pgbouncer in transaction mode. I was hoping there was a way to avoid disabling prepared statements as they surely give a performance boost, but I'll try it.

**Comment by dvarrazzo** [comment]
Prepared statements don't work with pgbouncer. Unfortunately pgbouncer doesn't advertise its presence, in which case we would be able to disable them automatically. See https://github.com/pgbouncer/pgbouncer/issues/653 for details.

**Comment by jfilby** [comment]
I've tried it but I still get the same error. Here's my connect code, which includes the setting to turn off prepared statements:

```
    db = psycopg.connect(f"host={DB_HOST} port={DB_PORT} dbname={DB_NAME} " +
                         f"user={DB_USERNAME} password={DB_PASSWORD}",
                         prepare_threshold = None)
```

**Comment by dvarrazzo** [comment]
That's because `prepare_threashold` as a connection parameter will be only available in psycopg 3.1. It should throw an exception, but, because the value is None, it gets ignored altogether. I think th should be addressed.

Please try to set `db.prepared_threshold = None` after the connection has been created.

**Comment by jfilby** [comment]
Thanks this has tested as working with your recommendation.
